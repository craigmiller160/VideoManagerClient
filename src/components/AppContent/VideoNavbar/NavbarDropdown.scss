
.NavbarDropdown {
  .dropdown {
    background-color: #343a40;
    border: none;

    .dropdownItem {
      padding: .25rem 1.5rem;
    }
  }
}