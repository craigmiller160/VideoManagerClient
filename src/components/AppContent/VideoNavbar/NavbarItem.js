import React from 'react';
import PropTypes from 'prop-types';
import { NavItem, NavLink as BootLink } from 'reactstrap';
import { NavLink } from 'react-router-dom';
import classes from './NavbarItem.scss';
import useReactRouter from 'use-react-router';

const NavbarItem = (props) => {
    const { history: { pathname } } = useReactRouter();
    const {
        id,
        to,
        text,
        exact,
        isLink,
        onClick
    } = props;

    return (
        <NavItem className={ classes.NavbarItem } active={ to && pathname === to }>
            <BootLink
                tag="div"
                id={ id }
                className={ classes['use-pointer'] }
                onClick={ onClick }
            >
                {
                    isLink &&
                    <NavLink
                        to={ to }
                        activeClassName={ classes.active }
                        className={ classes.link }
                        exact={ exact }
                    >
                        { text }
                    </NavLink>
                }
                {
                    !isLink && text
                }
            </BootLink>
        </NavItem>
    );
};
NavbarItem.propTypes = {
    id: PropTypes.string,
    to: PropTypes.string,
    text: PropTypes.string,
    exact: PropTypes.bool,
    isLink: PropTypes.bool,
    onClick: PropTypes.func
};
NavbarItem.defaultProps = {
    exact: false,
    isLink: false
};

export default NavbarItem;
