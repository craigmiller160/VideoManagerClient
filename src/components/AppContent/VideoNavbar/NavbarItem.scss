@import './navbar';

.NavbarItem {
  .link {
    @include link;
  }
}
