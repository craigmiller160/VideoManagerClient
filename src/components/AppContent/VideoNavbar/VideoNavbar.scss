.VideoNavbar {
    border-bottom: 1px solid lightgray;
    .use-pointer {
        cursor: pointer;
    }
}