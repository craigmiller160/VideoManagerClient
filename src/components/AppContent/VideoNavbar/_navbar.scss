@mixin link {
  color: rgba(255,255,255,.5);
  &:hover {
    color: white;
    text-decoration: none;
  }
  &.active {
    color: white;
  }
}
