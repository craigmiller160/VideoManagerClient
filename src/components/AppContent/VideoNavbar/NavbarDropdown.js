/*
 *     VideoManagerClient
 *     Copyright (C) 2020 Craig Miller
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, either version 3 of the License, or
 *     (at your option) any later version.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import React from 'react';
import { DropdownMenu, DropdownToggle, UncontrolledDropdown } from 'reactstrap';
import { shallowEqual, useDispatch, useSelector } from 'react-redux';
import useReactRouter from 'use-react-router';
import classes from './NavbarDropdown.scss';
import NavbarItem from './NavbarItem';
import { logout } from '../../../store/auth/auth.actions';

// TODO delete this, it's unnecessary

const NavbarDropdown = () => {
    const dispatch = useDispatch();
    const { location } = useReactRouter();
    const { firstName, lastName } = useSelector(state => state.auth.userDetails, shallowEqual);

    const toggleClasses = [];
    if (location.pathname === '/profile') {
        toggleClasses.push('active');
    }

    const fullName = `${firstName} ${lastName}`;

    return (
        <UncontrolledDropdown nav inNavbar className={ classes.NavbarDropdown }>
            <DropdownToggle id="vm-navbar-dropdown-toggle" nav caret className={ toggleClasses.join(' ') }>{ fullName }</DropdownToggle>
            <DropdownMenu id="vm-navbar-dropdown" right className={ classes.dropdown }>
                <NavbarItem
                    id="userProfileLink"
                    text="Profile"
                    className={ classes.dropdownItem }
                    isLink
                    exact
                    to="/profile"
                />
                <NavbarItem
                    id="logoutLink"
                    onClick={ () => dispatch(logout()) }
                    className={ classes.dropdownItem }
                    text="Logout"
                />
            </DropdownMenu>
        </UncontrolledDropdown>
    );
};

export default NavbarDropdown;
