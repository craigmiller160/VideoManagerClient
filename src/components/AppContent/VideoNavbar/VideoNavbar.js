import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Collapse, Container, Nav, Navbar, NavbarBrand, NavbarToggler, NavItem, NavLink } from 'reactstrap';
import * as classes from './VideoNavbar.scss'
import { Link } from 'react-router-dom';

class VideoNavbar extends Component {

    state = {
        isOpen: false
    };

    toggle = () => {
        this.setState((prevState) => {
            return {
                ...prevState,
                isOpen: !prevState.isOpen
            }
        });
    };

    handleScanDirectory = () => {
        this.props.startFileScan();
    };

    render() {
        const { isOpen } = this.state;

        return (
            <Navbar className={ classes.VideoNavbar } color="dark" dark expand="md">
                <Container>
                    <NavbarBrand tag={ Link } to="/">Video Manager</NavbarBrand>
                    <NavbarToggler onClick={ this.toggle } />
                    <Collapse isOpen={ isOpen } navbar>
                        <Nav className="ml-auto" navbar>
                            <NavItem>
                                <NavLink
                                    id="scanDirectory"
                                    className={ classes['use-pointer'] }
                                    onClick={ this.handleScanDirectory }>
                                    Scan Directory
                                </NavLink>
                            </NavItem>
                        </Nav>
                    </Collapse>
                </Container>
            </Navbar>
        );
    }

}

VideoNavbar.propTypes = {
    startFileScan: PropTypes.func
};

export default VideoNavbar;