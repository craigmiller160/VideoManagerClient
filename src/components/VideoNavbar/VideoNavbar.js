import React, { Component } from 'react';
import {
    Nav,
    Navbar,
    NavbarBrand,
    NavItem,
    NavbarToggler,
    Collapse,
    NavLink
} from 'reactstrap';
import * as classes from './VideoNavbar.scss';

class VideoNavbar extends Component {

    state = {
        isOpen: false
    };

    toggle = () => {
        this.setState((prevState) => {
            return {
                ...prevState,
                isOpen: !prevState.isOpen
            }
        });
    };

    render() {

        return (
            <Navbar className={classes.VideoNavbar} color="light" light expand="md">
                <NavbarBrand href="/">Video Manager</NavbarBrand>
                <NavbarToggler onClick={this.toggle} />
                <Collapse isOpen={this.state.isOpen} navbar>
                    <Nav className="ml-auto" navbar>
                        <NavItem>
                            <NavLink>Scan Directory</NavLink>
                        </NavItem>
                    </Nav>
                </Collapse>
            </Navbar>
        );
    }

}

export default VideoNavbar;