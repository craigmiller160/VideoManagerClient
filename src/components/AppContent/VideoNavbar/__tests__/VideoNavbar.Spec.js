import React from 'react';
import { mount } from 'enzyme';
import VideoNavbar from '../VideoNavbar';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';

const mountComponent = (props) => {
    props = props || {};
    return mount(
        <MemoryRouter initialEntries={ ['/'] }>
            <VideoNavbar { ...props } />
        </MemoryRouter>
    );
};

describe('VideoNavbar', () => {
    it('renders successfully', () => {
        const component = mountComponent();
        expect(toJson(component.find('VideoNavbar'))).toMatchSnapshot();
    });

    it('toggles the collapse open and closed', () => {
        const component = mountComponent();
        const navbar = component.find('VideoNavbar');
        const toggle = component.find('VideoNavbar NavbarToggler');

        expect(navbar.state()).toEqual(expect.objectContaining({
            isOpen: false
        }));
        toggle.simulate('click');
        expect(navbar.state()).toEqual(expect.objectContaining({
            isOpen: true
        }));
    });

    it('Starts file scan', () => {
        const startFileScan = jest.fn();
        const component = mountComponent({ startFileScan });
        const scanDirectoryBtn = component.find('NavLink#scanDirectory');
        scanDirectoryBtn.simulate('click');
        expect(startFileScan).toHaveBeenCalledTimes(1);
    });
});