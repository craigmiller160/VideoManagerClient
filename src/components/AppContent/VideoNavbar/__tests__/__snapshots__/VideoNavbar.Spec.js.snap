// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoNavbar renders successfully 1`] = `
<VideoNavbar>
  <Navbar
    color="dark"
    dark={true}
    expand="md"
    tag="nav"
  >
    <nav
      className="navbar navbar-expand-md navbar-dark bg-dark"
    >
      <Container
        tag="div"
      >
        <div
          className="container"
        >
          <NavbarBrand
            tag={[Function]}
            to="/"
          >
            <Link
              className="navbar-brand"
              replace={false}
              to="/"
            >
              <a
                className="navbar-brand"
                href="/"
                onClick={[Function]}
              >
                Video Manager
              </a>
            </Link>
          </NavbarBrand>
          <NavbarToggler
            onClick={[Function]}
            tag="button"
            type="button"
          >
            <button
              className="navbar-toggler"
              onClick={[Function]}
              type="button"
            >
              <span
                className="navbar-toggler-icon"
              />
            </button>
          </NavbarToggler>
          <Collapse
            appear={false}
            enter={true}
            exit={true}
            in={false}
            isOpen={false}
            mountOnEnter={false}
            navbar={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={350}
            unmountOnExit={false}
          >
            <Transition
              appear={false}
              enter={true}
              exit={true}
              in={false}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={350}
              unmountOnExit={false}
            >
              <div
                className="collapse navbar-collapse"
                style={Object {}}
              >
                <Nav
                  className="ml-auto"
                  navbar={true}
                  tag="ul"
                  vertical={false}
                >
                  <ul
                    className="ml-auto navbar-nav"
                  >
                    <NavItem
                      tag="li"
                    >
                      <li
                        className="nav-item"
                      >
                        <NavLink
                          onClick={[Function]}
                          tag="a"
                        >
                          <a
                            className="nav-link"
                            onClick={[Function]}
                          >
                            Scan Directory
                          </a>
                        </NavLink>
                      </li>
                    </NavItem>
                  </ul>
                </Nav>
              </div>
            </Transition>
          </Collapse>
        </div>
      </Container>
    </nav>
  </Navbar>
</VideoNavbar>
`;
